### PR TITLE
Deprecate EXTERNAL variables

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

**Comment by alkino** [comment]
Should we keep `inst` for those variables? Should we keep them in `codegen_float_variables`?

File: src/language/nmodl.yaml [hunk 1]
```diff
@@ -1770,7 +1770,7 @@
                             type: ExternVar
                             vector: true
                             separator: ", "
-                      brief: "Represents EXTERNAL statement in NMODL"
+                      brief: "This feature is DEPRECATED. Represents EXTERNAL statement in NMODL"
```
Comment (pramodk) [inline src/language/nmodl.yaml:1773]:
```suggestion
                      brief: "This construct is deprecated and no longer supported in the NMODL"
```

File: src/codegen/codegen_compatibility_visitor.hpp [hunk 2]
```diff
@@ -100,10 +100,17 @@ class CodegenCompatibilityVisitor: public visitor::AstVisitor {
...
         auto real_type_block = std::dynamic_pointer_cast<T>(ast_node);
-        return fmt::format("\"{}\" {}construct found at [{}] is not handled\n",
-                           ast_node->get_node_name(),
-                           real_type_block->get_nmodl_name(),
-                           real_type_block->get_token()->position());
+        auto token = real_type_block->get_token();
+        try {
+            return fmt::format("\"{}\" {}construct found at [{}] is not handled\n",
+                               ast_node->get_node_name(),
+                               real_type_block->get_nmodl_name(),
```
Comment (pramodk) [inline src/codegen/codegen_compatibility_visitor.hpp:107]: just curious - which of the `get_node_name()` result into exception here?

Comment (alkino) [reply src/codegen/codegen_compatibility_visitor.hpp:107]: `get_node_name()` has meaning only *named* objects. Just like a function.

**Review by pramodk — COMMENTED** [review]
apart from a single comment, LGTM!

File: test/unit/codegen/codegen_compatibility_visitor.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+using namespace visitor;
+using namespace codegen;
+
+using nmodl::parser::NmodlDriver;
+
+/// Return true if it failed and false otherwise
+bool runCompatibilityVisitor(const std::string& nmodl_text) {
+    const auto& ast = NmodlDriver().parse_string(nmodl_text);
+    SymtabVisitor().visit_program(*ast);
+    PerfVisitor().visit_program(*ast);
```
[the remark this answers is not in the document]
Comment (alkino) [reply test/unit/codegen/codegen_compatibility_visitor.cpp:29]: `PerfVisitor` is responsible of `write_count` so yes

**Review by pramodk — APPROVED** [review]
LGTM

**Pull request merged**